packages/expo-cli/src/commands/publish/publishAsync.ts

```diff
@@ -150,6 +154,25 @@ function assertValidReleaseChannel(releaseChannel?: string): void {
   }
 }
 
+function assertExpoUpdatesInstalled(projectDir: string, sdkVersion?: string): void {
+  // before sdk 44, expo-update was included in with the expo module
+  if (sdkVersion && semver.lt(sdkVersion, '44.0.0')) {
+    return;
+  }
+
+  const packageJsonPath = resolveFrom.silent(projectDir, './package.json');
+  if (packageJsonPath) {
+    const expoPackageJson = JsonFile.read(packageJsonPath, { json5: true });
+    if (expoPackageJson.dependencies && (expoPackageJson.dependencies as any)['expo-updates']) {
+      return;
+    }
+  }
```

Comment:
we have existing code for handling reading the package.json - in the `actionAsync` function above the `pkg` is returned from `getConfig`, for example. you could pass that pkg into this function instead of projectDir.

also, checking `dependencies` may not be a completely dependable way of determining whether the library is installed -- maybe it's accidentally installed as a dev dependency, in which case it would still work fine. there may be some edge cases in which the user is depending on expo-updates via a transitive dep (unlikely but with enough users is bound to happen) or they use something like nx for their monorepo, which i believe centralizes dependencies in a monorepo root package.json. see next comment for follow up on this point.

Reply:
Referring to `expo-dev-client` detecting in eas, that's simply done by finding `expo-dev-client/package.json` beneath the project root (so in node_modules): https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/build/utils/devClient.ts#L91-L99